components/engine_rocks/src/misc.rs

```diff
@@ -19,17 +77,60 @@ impl MiscExt for RocksEngine {
         Ok(self.as_inner().flush_cf(handle, sync)?)
     }
 
-    fn delete_files_in_range_cf(
+    fn delete_all_in_range_cf(
         &self,
         cf: &str,
+        strategy: DeleteStrategy,
         start_key: &[u8],
         end_key: &[u8],
-        include_end: bool,
-    ) -> Result<()> {
-        let handle = util::get_cf_handle(self.as_inner(), cf)?;
-        Ok(self
-            .as_inner()
-            .delete_files_in_range_cf(handle, start_key, end_key, include_end)?)
+    ) -> Result<usize> {
+        let mut wb = self.write_batch();
+        let start = KeyBuilder::from_slice(start_key, 0, 0);
+        let end = KeyBuilder::from_slice(end_key, 0, 0);
+        let mut opts = IterOptions::new(Some(start), Some(end), false);
+        if self.is_titan() {
+            // Cause DeleteFilesInRange may expose old blob index keys, setting key only for Titan
+            // to avoid referring to missing blob files.
+            opts.set_key_only(true);
+        }
+        let mut count = 0;
+
+        match strategy {
+            DeleteStrategy::DeleteFiles => {
+                let handle = util::get_cf_handle(self.as_inner(), cf)?;
+                self.as_inner()
+                    .delete_files_in_range_cf(handle, start_key, end_key, false)?;
+                Ok(0)
+            }
+            DeleteStrategy::DeleteByRange => {
+                wb.delete_range_cf(cf, start_key, end_key)?;
+                self.write(&wb)?;
+                Ok(1)
+            }
+            DeleteStrategy::DeleteByKey => {
+                let mut it = self.iterator_cf_opt(cf, opts)?;
+                let mut it_valid = it.seek(start_key.into())?;
+                let mut opts = WriteOptions::new();
+                opts.set_sync(false);
+                while it_valid {
+                    wb.delete_cf(cf, it.key())?;
+                    if wb.count() >= Self::WRITE_BATCH_MAX_KEYS {
+                        self.write_opt(&wb, &opts)?;
+                        wb.clear();
+                    }
+                    count += 1;
+                    it_valid = it.next()?;
+                }
+                if wb.count() > 0 {
+                    self.write_opt(&wb, &opts)?;
+                }
+                self.sync_wal()?;
```

Comment:
is it necessary?

Reply:
Yes. All deleted key must be persisted.

Reply:
we didn't do that before. And we will clean the range again before ingest, so I don't think it's necessary

Reply:
No. We used to persist at every time write a writebatch into rocksdb.